security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/AuthenticationStrategy.scala

```diff
@@ -0,0 +1,31 @@
+package pl.touk.nussknacker.ui.security.oauth2
+
+import pl.touk.nussknacker.ui.security.api.AuthenticatedUser
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait AuthenticationStrategy[ProfileResponse] {
+
+  def authenticateUser(
+      accessTokenData: IntrospectedAccessTokenData,
+      getProfile: => Future[ProfileResponse]
+  )(implicit ec: ExecutionContext): Future[AuthenticatedUser]
+
+}
+
+object AuthenticationStrategy {
+
+  def getUserRoles(
```

Comment:
these two functions look like great candidates for the extension methods of the `OAuth2Configuration` class.

Reply:
I converted the first one to a normal method in `OAuth2Configuration` and inlined the usage of the second one because it was used only in one place.